### ISSUE TITLE
module install failed

### ISSUE THREAD

**Issue by bruno-jing** [sysrepo/sysrepo#1884]
hi，
After I installed the o-ran-usermgmt.yang model using sysrepoctl, there was a problem with the sysrepo database. The error message is displayed below. It seems that this model cannot be successfully installed.
[root@rc224 /usr/app/om]# sysrepoctl -l
[ERR]: Opening "/etc/sysrepo/data/o-ran-usermgmt.startup" file failed (No such file or directory).
[ERR]: Copying module "o-ran-usermgmt" data from <startup> to <running> failed.
sysrepoctl error: Failed to connect (Internal error)

**Comment by michalvasko** [comment]
Hi,
okay, can you please post the output of previous commands with `-v3`?

Regards,
Michal

**Comment by bruno-jing** [comment]
hi,
okay,I just used the -v3 parameter to display more information. Below is the information displayed during the installation process.

This is the information displayed by installing o-ran-usermgmt using the sysrepoctl tool:
[root@rc224 /o-ran-modules]# sysrepoctl -i o-ran-usermgmt.yang -v3
[INF]: Applying scheduled changes.
[INF]: No scheduled changes.
[INF]: Module "o-ran-usermgmt" scheduled for installation.

This is the information displayed in the module list after installation. It looks like the module has been installed:
o-ran-module-cap           | 2019-07-03 | I     | root:root | 600         |            | LAA                                
o-ran-operations           | 2019-07-03 | I     | root:root | 600         |            |                                                       
o-ran-software-management  | 2019-07-03 | I     | root:root | 600         |            |                                                          
o-ran-transceiver          | 2019-07-03 | I     | root:root | 600         |            |                                                           
o-ran-usermgmt             |            |  N    |           |             |            |  

After the installation, I restarted the system and used sysrepoctl again to view the module list information. At this time, the database was abnormal：
[root@rc224 /]# sysrepoctl -l -v3
[INF]: Applying scheduled changes.
[WRN]: Must condition "user/enabled='true'" not satisfied. (path: /o-ran-usermgmt:users)
[WRN]: At least one account needs to be enabled. (path: /o-ran-usermgmt:users)
[WRN]: Failed to apply some changes, leaving all changes scheduled.
[ERR]: Opening "/etc/sysrepo/data/o-ran-usermgmt.startup" file failed (No such file or directory).
[ERR]: Copying module "o-ran-usermgmt" data from <startup> to <running> failed.
sysrepoctl error: Failed to connect (Internal error)

I'm not sure if there is any conflict between the module itself and sysrepo.
Regards,
Bruno

**Comment by michalvasko** [comment]
Hi Bruno,
the internal error should be fixed now, that was not supposed to happen. However, that is not the problem here.

You are installing a module, for which empty data are not valid. So *sysrepo* cannot install it this way because it would cause the datastore to be invalid. Instead, you need to also supply the initial configuration that will make the module valid. Do that using `sysrepocfg -W`, for example. Also note that `N` flag next to the module means it was __not__ installed and is only scheduled to be so. The reason why it failed to be installed was printed.

Regards,
Michal

**Comment by bruno-jing** [comment]
Hi
Can you elaborate on how this module that needs to be configured is installed, or there is such an installation tutorial，thanks

Regrads,
Bruno

**Comment by michalvasko** [comment]
Hi,
well, you should find all the general information in the [documentation](https://netopeer.liberouter.org/doc/sysrepo/master/schema.html), I see no reason to repeat that. I have also told you how to set the initial configuration so I am not sure what else to add.

Regards,
Michal

**Comment by sjd-xlnx** [comment]
Forgive me for posting on this "issue" which has been closed for some time, but I found the same problem and this issue came up when "googling" it, and I have a solution that I am posting here in case anyone else sees the problem.

The "problem" is that o-ran-usermgmt.yang has a "must" condition that means at least one user account must be enabled. The problem being how to configure the account before you have the yang module installed (kind of "chicken and egg" problem). The previous poster was on the right track, but was using sysrepocfg, which unfortunately doesn't work since we didn't manage to install the o-ran-usermgmt.yang yet. A better solution is to use the "-I, --init-data" option of sysrepctl.

```sysrepoctl --install o-ran-usermgmt.yang -v3 --init-data o-ran-usermgmt.xml```

Where o-ran-usermgmt.xml is as posted above. This installs and initialized the data in one step, so chicken and egg at the same time!
